app.go

```diff
@@ -45,6 +45,55 @@ type LedgerOasis struct {
 	version VersionInfo
 }
 
+// Displays existing Ledger Oasis apps by address
+func ListOasisDevices(path []uint32) {
+	for _, d := range ledger_go.FindLedgerDevices() {
+		device, err := d.Open()
+		if err != nil {
+			continue
+		}
+		ledgerAPI := ledger_go.NewLedger(device)
+		app := LedgerOasis{ledgerAPI, VersionInfo{}}
+		defer app.Close()
+
+		appVersion, err := app.GetVersion()
+		if err != nil {
+			continue
+		}
+
+		_, address, err := app.GetAddressPubKeyEd25519(path)
+		if err != nil {
+			continue
+		}
+
+		fmt.Printf("============ Device found\n")
+		fmt.Printf("Oasis App Version : %x\n", appVersion)
+		fmt.Printf("Oasis App Address : %s\n", address)
+	}
+}
+
+// ConnectLedgerOasisApp connects to Oasis app based on address
+func ConnectLedgerOasisApp(seekingAddress string, path []uint32) (*LedgerOasis, error) {
+	for _, d := range ledger_go.FindLedgerDevices() {
+		device, err := d.Open()
+		if err != nil {
+			continue
+		}
+		ledgerAPI := ledger_go.NewLedger(device)
+		app := LedgerOasis{ledgerAPI, VersionInfo{}}
+
+		_, address, err := app.GetAddressPubKeyEd25519(path)
```

Comment:
What kind of address are you comparing here? This should be something that is independent of the derivation path and only dependent on the actual device (like a device ID).

Reply:
I responded this in the integration PR, but from conversations with @jleni , there is not reliable device ID, so we are using address to identify devices.

Reply:
Correct. Devices themselves do not have a serial number and are not identifiable.